modules/phase_field/include/kernels/CHInterface.h

```diff
@@ -28,24 +30,33 @@ class CHInterface : public Kernel
 protected:
   virtual Real computeQpResidual();
   virtual Real computeQpJacobian();
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
 
 private:
   std::string _kappa_name;
   std::string _mob_name;
-  std::string _Dmob_name;
-  std::string _grad_mob_name;
 
+  ///Material property values
   const MaterialProperty<Real> & _kappa;
   const MaterialProperty<Real> & _M;
-  bool _has_MJac;
-  const MaterialProperty<Real> * _DM;
-  const MaterialProperty<RealGradient> & _grad_M;
-  const MaterialProperty<RealGradient> * _Dgrad_Mnp;
-  const MaterialProperty<Real> * _Dgrad_Mngp;
+  const MaterialProperty<Real> & _dMdc;
+  const MaterialProperty<Real> & _d2Mdc2;
 
+  ///Variables for second order derivatives
   VariableSecond & _second_u;
   VariableTestSecond & _second_test;
   VariablePhiSecond & _second_phi;
+
+  ///Number of variables
+  unsigned int _nvar;
+
+  ///Material property vectors
```

Comment:
@andrsd would call this _Captain Obvious_... ...me, too.

By the way you can make Doxygen doc comments apply to multiple members by wrapping them with ```@{ @}``` like this:
```
///@{
/// Mobility derivatives
std::vector<const MaterialProperty<Real> *> _dMdarg;
std::vector<const MaterialProperty<Real> *> _d2Mdcdarg;
std::vector<std::vector<const MaterialProperty<Real>* > > _d2Mdargdarg;
///@}
```